### ISSUE TITLE
how to make you demo gif style?

### ISSUE THREAD

**Issue by luoolu** [ChristophReich1996/Yeast-in-Microstructures-Dataset#1]
gif making

**Comment by ChristophReich1996** [comment]
Hi @luoolu, the gif/video was produced by concatenating the image (w/o the label) and the overlay (image + label) at different indexes. You can use the following script:

```python
import torch
import torchvision
from torch import Tensor


def main() -> None:
    # Make some images (here you need to load the two images)
    first_img: Tensor = torch.linspace(0, 255, 768)[None, None].repeat(3, 256, 1)
    second_img: Tensor = torch.zeros(3, 256, 768)
    # Init tensor to store video shape is [T, 3, H, W]
    video: Tensor = torch.empty(768 // 8 + 2, 3, 256, 768)
    # Make video frames as the concatenation of the first and second frame
    for frame_index, index in enumerate(list(range(0, 768 + 1, 16)) + list(reversed(range(0, 768 + 1, 16)))):
        print(frame_index)
        # In this case we just take the first image
        if index == 0:
            video[frame_index] = first_img
        # In this case we just take the second image
        if index == 768:
            video[frame_index] = second_img
        # In this case we concatenate both images
        else:
            video[frame_index] = torch.cat((first_img[..., :-(1 + index)], second_img[..., -(1 + index):]), dim=2)
    # Save video, we need to permute the video since write_video uses the format [T, H, W, C]
    torchvision.io.video.write_video("animation.mp4", video.permute(0, 2, 3, 1), fps=30)


if __name__ == '__main__':
    main()
```

You can either use the animation in the `.mp4` format or use (for instance) [FFmpeg](https://ffmpeg.org) to convert the video into a `.gif`.

For generating the overlay visualization please refer to the provided [visualization script](https://github.com/ChristophReich1996/Yeast-in-Microstructures-Dataset/blob/main/example_vis.py).

PS: If you find this helpful consider staring this repo :)

Best,
Christoph
